packages/engine/src/Blocks.js

```diff
@@ -91,8 +91,13 @@ class Blocks {
       block.styleEval = {};
       block.validationEval = {};
       block.visibleEval = {};
-
-      block.meta = this.context._internal.lowdefy._internal.blockComponents[block.type].meta;
+      try {
```

Comment:
Should engine be the thing throwing here? I guess we create the context before we start rendering, so this is the first place it throws. But build is also not supposed to build if a block type doesn't exist.

Is this the right error message? Why were you running into this error without build throwing?

One other thing that could happen here is a bad block that does not have a meta definition, or a plugin that declares a block that does not exist.

Reply:
True, those checks should happen else where as well, however, this error is good to indicate that there is an issue with the block, instead of simply throwing can not read meta of undefined. Perhaps the message should only read: `Invalid block definition for type ${type} at block ${blockId}`?

Reply:
Happy with this message, can we add it in the next PR? Or this oneif you make other changes.